langchain/src/chat_models/openai.ts

```diff
@@ -548,7 +648,23 @@ export class ChatOpenAI<
           message.name !== undefined
             ? tokensPerName + (await this.getNumTokens(message.name))
             : 0;
-        const count = textCount + tokensPerMessage + roleCount + nameCount;
+        let count = textCount + tokensPerMessage + roleCount + nameCount;
+
+        const openAIMessage = messageToOpenAIMessage(message);
+        if (openAIMessage.role === "function") {
+          count -= 2;
```

Comment:
Can you add a comment for why this is?

Reply:
It is taken from https://github.com/hmarr/openai-chat-tokens/blob/main/src/index.ts, I'll add it into a comment